### PR TITLE
docs/customization: Clarify "day 1" MachineConfigs are supported

### PR DESCRIPTION
**Pull request by cgwalters**
We need to support this for cases such as `nosmt` and others.
Clarify that it's supported and elaborate on that a bit.

### CODE REVIEW

File: docs/user/customization.md [hunk 1]
```diff
@@ -226,11 +226,12 @@ The `manifest-templates` target will output the unrendered manifest templates in
 **IMPORTANT**:
 
 - These customizations require using the `manifests` target that does not provide compatibility guarantees.
-- This can affect upgradability of your cluster as the `machine-config-operator` can mark clusters tainted when user defined [MachineConfig][machine-config] objects are present in the cluster.
```
Comment (cgwalters) [inline docs/user/customization.md]: This bit was always a lie.

**Comment by abhinavdahiya** [comment]
Using nosmt as example when it's supported as install-config options creates more confusion.

**Comment by cgwalters** [comment]
> Using nosmt as example when it's supported as install-config options creates more confusion.

Hmm...I guess bigger picture I'd like to deprecate the `hyperthreading` flag in favor of this, as it's clearer to admins what's going on.  (OTOH, there's implicit validation in specifying `hyperthreading`, you can't typo the kernel argument as `nosmy` or whatever...)

Anyways, I guess I can mention multiple, say `nosmt` and  e.g. `rd.neednet ip=...` for cases like NBDE/Tang that require networking in the initrd?

**Comment by abhinavdahiya** [comment]
> > Using nosmt as example when it's supported as install-config options creates more confusion.
> 
> Hmm...I guess bigger picture I'd like to deprecate the `hyperthreading` flag in favor of this, as it's clearer to admins what's going on. (OTOH, there's implicit validation in specifying `hyperthreading`, you can't typo the kernel argument as `nosmy` or whatever...)

configuring hyperthreading was important configurable to be exposed in an API even though it can be done by manifests target.

the manifest target continues to be best-effort supported in terms of compatibility, and is most definitely only advised for advanced users, even though day-2 machine configs are supported by mco. Therefore i would request we provide other examples.

> 
> Anyways, I guess I can mention multiple, say `nosmt` and e.g. `rd.neednet ip=...` for cases like NBDE/Tang that require networking in the initrd?

File: docs/user/customization.md [hunk 2]
```diff
@@ -226,11 +226,12 @@ The `manifest-templates` target will output the unrendered manifest templates in
...
 
 In most cases, user applications should be run on the cluster via Kubernetes workload objects (e.g. DaemonSet, Deployment, etc). For example, DaemonSets are the most stable way to run a logging agent on all hosts. However, there may be some cases where these workloads need to be executed prior to the node joining the Kubernetes cluster. For example, a compliance mandate like "the user must run auditing tools as soon as the operating system comes up" might require a custom systemd unit for an auditing container in the Ignition config for some or all nodes.
 
-The configuration of machines in OpenShift is controlled using `MachineConfig` objects and what configuration is applied to a machine in the OpenShift cluster is based on the [MachineConfigPool][machine-config-pool] objects. To allow customization of machine configuration which is not possible as Day 2 operation, the installer allows users to bring their own custom `MachineConfig` objects.
+Further, some aspects of RHEL CoreOS machines (usually kernel arguments such as `nosmt` for disabling hyperthreading) may need to be configured before user workloads land on a system.
```
Comment (kikisdeliveryservice) [inline docs/user/customization.md]: Maybe move the sentence from below: "For these "day 1" cases, `MachineConfig` objects can be provided as additional manifests.` to after this sentence?

Comment (kikisdeliveryservice) [reply docs/user/customization.md]: If you do use nosmt, maybe link to the MCO's docs : https://github.com/openshift/machine-config-operator/blob/5795bb4f2f7b3adf3cf3dfb4100774244a1b8f99/docs/MachineConfiguration.md#nosmt

**Comment by abhinavdahiya** [comment]
ping @cgwalters

**Pull request merged**